PIL/JpegImagePlugin.py

```diff
@@ -713,8 +713,8 @@ def _save_cjpeg(im, fp, filename):
 # Factory for making JPEG and MPO instances
 def jpeg_factory(fp=None, filename=None):
     im = JpegImageFile(fp, filename)
-    mpheader = im._getmp()
     try:
+        mpheader = im._getmp()
```

Comment:
I agree with this change, but I think that the errors should actually be SyntaxErrors, since that's what we're using for 'not this type of image' elsewhere in the code.